unikernel.ml

```diff
@@ -973,62 +1009,254 @@ struct
+                      Builder_web.build_of_json
+                        (Yojson.Basic.from_string response_body)
+                    with
+                    | Error (`Msg err) ->
+                        Logs.err (fun m ->
+                            m
+                              "JSON parsing of the latest build of %s from \
+                               Builder_web failed with error: %s"
+                              name err);
+                        Middleware.redirect_to_error
+                          ~data:
+                            (`String
+                              ("An error occured while parsing the json of the \
+                                latest build from Builder_web. The error \
+                                gotten is: " ^ err))
+                          ~title:(name ^ "update Error") ~api_meth:false
+                          `Internal_server_error reqd ()
+                    | Ok latest_job_data -> (
+                        if
+                          String.equal latest_job_data.uuid
+                            current_job_data.uuid
+                        then (
+                          Logs.info (fun m ->
+                              m
+                                "There is no new update of %s found with uuid  \
+                                 %s"
+                                name latest_job_data.uuid);
+                          Middleware.redirect_to_page
+                            ~path:
+                              ("/unikernel/info/"
+                              ^ Option.value ~default:""
+                                  (Vmm_core.Name.name unikernel_name))
+                            reqd
+                            ~msg:("There is no update of " ^ name ^ " found.")
+                            ())
+                        else
+                          Builder_web.send_request http_client
+                            (Builder_web.base_url ^ "/compare/"
+                           ^ current_job_data.uuid ^ "/" ^ latest_job_data.uuid
+                           ^ "")
+                          >>= function
+                          | Error (`Msg err) ->
+                              Logs.err (fun m ->
+                                  m
+                                    "Builder_web: Error while fetching the \
+                                     diff between the current and latest build \
+                                     info of %s with error: %s"
+                                    name err);
+                              Middleware.redirect_to_error
+                                ~data:
+                                  (`String
+                                    ("An error occured while fetching the diff \
+                                      between the latest and the current build \
+                                      information from Builder_web. The error \
+                                      gotten is: " ^ err))
```

Comment:
as above -- I don't think that a user of mollymawk understands what `Builder_web` is -- but providing the URL may give some insights.